src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrOfStates = [];
+
+  arrOfStates.push(state);
+
+  for (const action of actions) {
+    const stateCopy = { ...arrOfStates[arrOfStates.length - 1] };
+
+    if (action.type === 'addProperties') {
```

Comment:
try use switch case when you want to check a value for multiple possible results